pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -73,13 +82,25 @@ func validateMachinePool(client *Client, ic *types.InstallConfig, fldPath *field
 			zones = zones[:3]
```

Comment:
Cannot assume 3 here.
1. This should use the number of control plane replicas.
2. There may be less zones than there are control plane replicas.

Reply:
Ok, get it